community/index/src/test/java/org/neo4j/index/internal/gbptree/GBPTreeConcurrencyITBase.java

```diff
@@ -243,7 +243,7 @@ private void shouldReadCorrectlyWithConcurrentUpdates( TestCoordinator testCoord
         private final double writePercentage;
         private final AtomicReference<ReaderInstruction> currentReaderInstruction;
         private final boolean partitionedSeek;
-        TreeSet<Long> readersShouldSee;
+        Set<Long> readersShouldSee;
```

Comment:
This change causes test flakiness due to this set being cloned on line 261 and an alternate constructor being resolved. This would need to be at least a SortedSet for this reason. Please check for other places where this may have occurred too.

Reply:
Yes. That's something I just learned, thanks for the hint.

I checked all places of `TreeSet` and this is the only place for such scenario.